crates/ethrpc/src/http.rs

```diff
@@ -53,6 +54,70 @@ impl HttpTransport {
     fn new_request(&self) -> (Client, Arc<Inner>) {
         (self.client.clone(), self.inner.clone())
     }
+
+    fn insert_call_metadata(call: &mut Call) {
+        if let Call::MethodCall(method_call) = call {
+            if let Some(request_id) = observe::request_id::get_task_local_storage() {
+                match &mut method_call.params {
+                    Params::Array(_) => {
+                        let original_params = Value::from(method_call.params.clone());
+                        method_call.params = Params::Map(
```

Comment:
Converts Array params type to Map to store the original data and metadata using different keys to parse them later.